### ISSUE TITLE
gRPC OAuth Support

### ISSUE THREAD

**Issue by BirknerAlex** [a85/POSTMan-Chrome-Extension#11769]
### Is there an existing request for this feature?

- [X] I have searched the existing issues for this feature request and I know that duplicates will be closed

### Is your feature request related to a problem?

Missing support for OAuth JWT tokens when using gRPC.

### Describe the solution you'd like

I would like to use Postman for making gRPC requests using a Bearer Token via OAuth 2.0. Currently only `API Key`, `Basic Auth` and `Bearer Token` is supported when using gRPC.

But since our API is using Keycloak JWT tokens which are expiring after 5 minutes its is nearly impossible to use Postman without copying a new token each 5 minutes to the client.

**Comment by flaxxed** [comment]
Yes, please!  I'm looking for the same type of support that exists for the OAuth 2.0 option for REST calls.

**Comment by edgaralves** [comment]
Share the same pain, would be great to have this feature

**Comment by ghost** [comment]
Yes please! In general, the whole automated authorization flow that the HTTPS requests offer, should also be available for the GRPC calls :raised_hands:

**Comment by gfiecht** [comment]
Yes, we are in the same situation and would highly appreciate this feature! 👍

**Comment by jan-wilhelm** [comment]
same exact situation here, would love any update on this!

**Comment by aidanjdonohue** [comment]
I would trade my first born son for this feature

**Comment by BirknerAlex** [comment]
We would still love to see that implemented, this is the only missing feature which prevents us from using Postman in the company. I mean you have it basically already implemented for REST APIs, why not for GRPC?

I thought asking after 2 years again should be fine.

**Comment by apoorvgupta03** [comment]
https://github.com/user-attachments/assets/e135da9c-f16b-4c14-9ff0-8b57011f02ad

Hey everyone, we've added support for OAuth2 authentication in gRPC requests. Please update your Postman application to the latest version and try sending your requests using OAuth2 authentication.

If you encounter any further issues, kindly report them here. Thank you for your patience and feedback.

**Comment by BirknerAlex** [comment]
Should it work also for server reflection requests? Because running I am running now `PostmanAgent 0.4.45, darwin-24.5.0 on x64` and getting the error "Details: Unsupported authorization type: oauth2", I was able to receive an valid OAuth token via Postman.

<img width="1025" alt="Image" src="https://github.com/user-attachments/assets/70a09f45-837e-4bd3-a26d-5fb5197435b7" />

It looks like something is missing?

**Comment by apoorvgupta03** [comment]
Hi @BirknerAlex,
Please update both your Postman app and Desktop Agent to the latest versions. OAuth2 should now work for server reflection requests as well.

**Comment by friedemannf** [comment]
Hey @apoorvgupta03, I'm actually running into some issues with OAuth 2.0 during gRPC requests:

While fetching the tokens works just fine, it doesn't seem to be appended correctly to the header.

Did a short screen recording here:

https://github.com/user-attachments/assets/e4e10cd6-b237-4761-8e20-acad75db6ca2

1. Selecting the type `Bearer Token` correctly sends the token as `authorization: Bearer <token>`
2. Selecting the type `OAuth 2.0` and using one of the available tokens (or setting the token manually) does only send the header `authorization: <prefix>` but not the token itself.

This is on version `12.11.3`

**Comment by apoorvgupta03** [comment]
Hey @friedemannf,  thanks for the detailed report and the screen recording.

Could you please update Postman to the latest version (12.13.4) and let us know if the issue still reproduces consistently? If it does, it would really help if you could email us the [logs](https://support.postman.com/hc/en-us/articles/360025298633-How-to-get-logs-from-the-Postman-Desktop-app) at: [api-client@postman.com](mailto:api-client@postman.com), ideally captured right after a failing OAuth 2.0 gRPC invoke. Thanks!
